rasa/graph_components/validators/finetuning_validator.py

```diff
@@ -0,0 +1,341 @@
+    ) -> None:
+        """Validates whether we can finetune the NLU part when finetuning is enabled.
+
+        Args:
+            importer: a training data importer
+            config: the configuration (including the recipe)
+
+        Raises:
+            `InvalidConfigException` if there is a conflict
+        """
+        self._validate(importer=importer, config=config, nlu=True, core=False)
+
+    def validate_core_only(
+        self, importer: TrainingDataImporter, config: Dict[Text, Any]
+    ) -> None:
+        """Validates whether we can finetune the Core part when finetuning is enabled.
+
+        Args:
+            importer: a training data importer
+            config: the configuration (including the recipe)
+
+        Raises:
+            `InvalidConfigException` if there is a conflict
+        """
+        self._validate(importer=importer, config=config, nlu=False, core=True)
+
+    def validate(
+        self, importer: TrainingDataImporter, config: Dict[Text, Any],
+    ) -> None:
+        """Validates whether we can finetune Core and NLU when finetuning is enabled.
+
+        Args:
+            importer: a training data importer
+            config: the configuration (including the recipe)
+
+        Raises:
+            `InvalidConfigException` if there is a conflict
+        """
+        self._validate(importer, config, nlu=True, core=True)
+
+    def _validate(
+        self,
+        importer: TrainingDataImporter,
+        config: Dict[Text, Any],
+        nlu: bool = True,
+        core: bool = True,
+    ) -> None:
+        """Validate whether the finetuning setting conflicts with other settings.
+
+        Note that this validation always takes into account the configuration of
+        nlu *and* core part, while the validation of aspects of the domain and
+        the NLU training data only happen if we request to validate finetuning
+        with respect to NLU/Core models, respectively.
+
+        For more details, see docstring of this class.
+
+        Args:
+            importer: a training data importer
+            domain: the domain
+            config: the configuration (including the recipe)
+            nlu: set to `False` if NLU part should not be validated
+            core: set to `False` if Core part should not be validated
+        Raises:
+            `InvalidConfigException` if there is a conflict
+        """
+        finetuning = self._execution_context.is_finetuning
+
+        if finetuning and not self._fingerprints:
+            raise InvalidConfigException(
+                f"Finetuning is enabled but the {self.__class__.__name__} "
+                f"does not remember seeing a training run. Ensure that you have "
+                f"trained your model at least once (with finetuning disabled) "
+                f"and ensure that the  {self.__class__.__name__} is part of the "
+                f"training graph. "
+            )
+
+        # check the rasa version
```

Comment:
If a comment is needed, a nicely named method would be preferable in my opinion.